h3/src/qpack/decoder.rs

```diff
@@ -169,6 +169,41 @@ impl Decoder {
     }
 }
 
+// Decode a header bloc received on Request or Push stream. (draft: 4.5)
+pub fn decode_stateless<T: Buf>(buf: &mut T) -> Result<Vec<HeaderField>, Error> {
+    let (required_ref, _base) = HeaderPrefix::decode(buf)?.get(0, 0)?;
+
+    if required_ref > 0 {
+        return Err(Error::MissingRefs(required_ref));
+    }
+
+    let mut fields = Vec::new();
+    while buf.has_remaining() {
+        let field = match HeaderBlockField::decode(buf.bytes()[0]) {
```

Comment:
I think I understand why, but just to double check: this doesn't use `buf.get_u8()` because that advances the buffer and below might return an error about not enough bytes yet, right?

Reply:
We don't consume the first byte here because its first bits indicates the field type and the others are part of field data. So we leave advancing to the Header field decoder itself.